src/bun.js/module_loader.zig

```diff
@@ -2368,122 +2389,46 @@ pub const ModuleLoader = struct {
+                .source_url = specifier,
+                .hash = 0,
+                .tag = .esm,
+                .source_code_needs_deref = true,
+            },
+            .@"bun:internal-for-testing" => {
+                if (!Environment.isDebug) {
+                    if (!is_allowed_to_use_internal_testing_apis)
+                        return null;
+                }
+                return jsSyntheticModule(.InternalForTesting, specifier);
+            },
+            .@"bun:wrap" => .{
                 .allocator = null,
                 .source_code = String.init(Runtime.Runtime.sourceCode()),
                 .specifier = specifier,
                 .source_url = specifier,
                 .hash = Runtime.Runtime.versionHash(),
-            };
-        } else if (HardcodedModule.Map.getWithEql(specifier, bun.String.eqlComptime)) |hardcoded| {
-            Analytics.Features.builtin_modules.insert(hardcoded);
-
-            switch (hardcoded) {
-                .@"bun:main" => {
-                    return ResolvedSource{
-                        .allocator = null,
-                        .source_code = bun.String.createUTF8(jsc_vm.entry_point.source.contents),
-                        .specifier = specifier,
-                        .source_url = specifier,
-                        .hash = 0,
-                        .tag = .esm,
-                        .source_code_needs_deref = true,
-                    };
-                },
-
-                // Native modules
-                .bun => return jsSyntheticModule(.bun, specifier),
-                .@"node:buffer" => return jsSyntheticModule(.@"node:buffer", specifier),
-                .@"node:string_decoder" => return jsSyntheticModule(.@"node:string_decoder", specifier),
-                .@"node:module" => return jsSyntheticModule(.@"node:module", specifier),
-                .@"node:process" => return jsSyntheticModule(.@"node:process", specifier),
-                .@"node:tty" => return jsSyntheticModule(.@"node:tty", specifier),
-                .@"node:util/types" => return jsSyntheticModule(.@"node:util/types", specifier),
-                .@"node:constants" => return jsSyntheticModule(.@"node:constants", specifier),
-                .@"bun:jsc" => return jsSyntheticModule(.@"bun:jsc", specifier),
-                .@"bun:test" => return jsSyntheticModule(.@"bun:test", specifier),
-
-                .@"bun:internal-for-testing" => {
-                    if (!Environment.isDebug) {
-                        if (!is_allowed_to_use_internal_testing_apis)
-                            return null;
-                    }
+            },
+            inline else => |tag| jsSyntheticModule(@field(ResolvedSource.Tag, @tagName(tag)), specifier),
+        };
+    }
 
-                    return jsSyntheticModule(.InternalForTesting, specifier);
-                },
+    pub fn fetchBuiltinModule(jsc_vm: *VirtualMachine, specifier: bun.String) !?ResolvedSource {
```

Comment:
```suggestion
    pub fn fetchBuiltinModule(jsc_vm: *VirtualMachine, specifier: bun.String) ?ResolvedSource {
```

doesn't need to error anymore